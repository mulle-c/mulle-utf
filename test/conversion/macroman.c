#include <mulle-utf/mulle-utf.h>

#include <stdio.h>
#include <stdlib.h>



int  main()
{
   char            macos[ 256];
   int             n;
   unsigned int    m;
   char            buf[ 256 * 4] = { 0 };
   char            *end;
   char            *s;
   char            *t;
   int             i;
   unsigned int    off;

   for( n = 0, i = 32; i < 256; i++, n++)
      macos[ n] = i;

   end = _mulle_macroman_convert_to_utf8( macos, n, buf);

   // MEMO: the test output is wrong but I don't know why!
   off = 32;
   s   = buf;
   for(;;)
   {
      m = 16;
      t = mulle_utf8_skiputf32( s, &m);
      if( m != 16)
         break;

      printf( "%02x: %.*s\n", off, (int) (t - s), s);
      s    = t;
      off += 16;
   }

   if( end != s)
   {
      printf( "%02x: %.*s\n", off, (int) (end - s), s);
   }
   return( 0);
}

