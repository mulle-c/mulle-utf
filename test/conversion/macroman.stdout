20:  !"#$%&'()*+,-./
30: 0123456789:;<=>?
40: @ABCDEFGHIJKLMNO
50: PQRSTUVWXYZ[\]^_
60: `abcdefghijklmno
70: pqrstuvwxyz{|}~
80: ÄÅÇÉÑÖÜáàâäãåçéè
90: êëíìîïñóòôöõúùûü
a0: †°¢£§•¶ß®©™´¨≠ÆØ
b0: ∞±≤≥¥µ∂∑∏π∫ªºΩæø
c0: ¿¡¬√ƒ≈∆«»… ÀÃÕŒœ
d0: –—“”‘’÷◊ÿŸ⁄€‹›ﬁﬂ
e0: ‡·‚„‰ÂÊÁËÈÍÎÏÌÓÔ
f0: ÒÚÛÙıˆ˜¯˘˙˚¸˝˛ˇ
